### ISSUE TITLE
Support 32-bit R

### ISSUE THREAD

**Issue by smortaz** [Microsoft/RTVS#1250]
from a user on HN:

Any plans for a 32 bit version? We need our 32 bit R studio install to be able to run in parallel with this as we test this and we cannot switch Java versions from 32 to 64 bit which some pkgs we use require.

**Comment by jflam** [comment]
To future readers of this issue, my question to you is this: what are the scenarios where you absolutely need 32 bit R and where cross-process invocation (e.g., web socket, htttp etc.) would not solve your problem? This is an expensive feature for our test matrix. 

For now, closing this issue until we have more data.

**Comment by retrography** [comment]
The response is: Legacy environments, and that includes every single workstation in my school. Many organizations are holding back 64-bit Windows for compatibility purposes. Microsoft's Office365 website still recommends installing the 32-bit version of office as many 3rd-party tools are incompatible with the 64 version. I would gladly change this if I could, but I am pretty sure if the IS department is holding back, they have a good reason for it.

**Comment by AMarthallerMS** [comment]
There are some packages that don't work in 64 bit R.

serialized <- jsonlite::toJSON(filteredEvents)

Error in FUN(X[[i]], ...) : 
  no such symbol R_base64_decode in package C:/Users/###/Documents/R/win-library/3.2/jsonlite/libs/x64/jsonlite.dll

**Comment by AMarthallerMS** [comment]
Correction, my issue was a bad package install.
https://github.com/jeroenooms/jsonlite/issues/139

Of course the bad install was due to an orphaned Microsoft.R.Host.exe which I will open a separate issue for.

**Comment by JasonvanderStaaij** [comment]
To bad 32bit is not supported, why not support 32bit like visual studio. This amazing tool is installed (according to the documentation) in 

> %ProgramFiles(x86)%\Microsoft Visual Studio <VS version>\Common7\IDE\Extensions\Microsoft\R Tools for Visual Studio
which is the location for 32bit applications. If there are technical limitations, can you address them in more detail, so that the community might come up with solutions

![image](https://cloud.githubusercontent.com/assets/26189900/23581595/082fc8cc-0117-11e7-94c6-073dc4e6a22e.png)

**Comment by MikhailArkhipov** [comment]
The extension itself is 32-bit since it loaded into VS which is 32-bit. However, R runs out of process in Microsoft.R.Host.exe which is 64-bit.  The issue with 32-bit is not technical but rather time/resources constraint. Supporting multiple variants increases test matrix quite a bit. In remote-only case it would be possible to allow 32-bit OS since we would not be running R locally.

**Comment by MikhailArkhipov** [comment]
There may be difficulties running Visual Studio on 32-bit OS along with RTVS and R out of process. Although VS is indeed a 32-bit application, it requires fair amount of memory. On 32-bit machine Windows itself probably reserves 600-800 MB for OS needs, then VS may pretty easily consume another 1.2+ GB (and over 1.5 GB memory footprint is not unusual). On top of that we have to account for the R engine and its  hosting process. With just 4 GB memory available VS may end up pushing machine to its limits.

**Comment by JasonvanderStaaij** [comment]
Mikhail,
Thanks For the response. I understand the memory limits, which is beside the CPU limit of my current laptop a good reason for me to upgrade to a modern system.

**Comment by MikhailArkhipov** [comment]
This is unlikely to happen.
